apps/aecore/lib/aecore/chain/worker.ex

```diff
@@ -273,4 +309,31 @@ defmodule Aecore.Chain.Worker do
       blocks_acc
     end
   end
+
+  defp get_block_info_by_height(height, chain_hash \\ nil) do
+    begin_hash = if chain_hash == nil do top_block_hash() else chain_hash end
+    blocks_map = GenServer.call(__MODULE__, :blocks_map)
+    n = blocks_map[begin_hash].block.header.height - height
+    if n < 0 do
+      {:error, "Height higher then chain_hash height"}
+    else
+      blocks_map[get_nth_prev_hash(n, 0, begin_hash, blocks_map)]
+    end
+  end
+  
+  defp get_nth_prev_hash(0, _i, hash, blocks_map) do
+    hash
+  end
+
+  defp get_nth_prev_hash(n, i, hash, blocks_map) do
```

Comment:
write documentation to explain how this function works, or rewrite to be readable (no bit-magic)

Reply:
ok

Reply:
will you still work on this?